### DISCUSSION TITLE
Support ReGex patterns in namespace paths

### DISCUSSION THREAD

**Discussion by feds01** [RobinTail/zod-sockets#245] [Ideas]
Thanks for this wonderful library!

The Socket.IO library allows namespaces to have [dynamic paths](https://socket.io/docs/v4/namespaces/#dynamic-namespaces), specifically using ReGex. Is it currently possible to add namespaces in this library with dynamic paths, if not can this be a feature request?

**Comment by RobinTail** [comment]
That might be impossible to support, but I'll check

**Reply by feds01** [reply]
Perhaps its possible to use template literals instead, to make this more dynamic. I don't think TypeScript would allow arbitrary regular expressions as types (albeit would be a cool feature).

**Reply by RobinTail** [reply]
Yeah, I have some idea how to bypass that. Need some time to check it.
